pkg/splunk/client/azureblobclient.go

```diff
@@ -108,126 +203,242 @@ func NewAzureBlobClient(ctx context.Context, bucketName string, storageAccountNa
-	// This part need to be developed when the Azure blob rest handlers
-	// are built.
+	// Update httpRequest header with the HMAC256 signature
+	httpRequest.Header[headerAuthorization] = []string{authHeader}
 
-	// var err error
+	return nil
+}
 
-	// Enforcing minimum version TLS1.2
-	// tr := &http.Transport{
-	// 	TLSClientConfig: &tls.Config{
-	// 		MinVersion: tls.VersionTLS12,
-	// 	},
-	// }
-	// tr.ForceAttemptHTTP2 = true
-	// httpClient := http.Client{Transport: tr}
+// Update http request header with IAM info
+func updateAzureHttpRequestHeaderWithIAM(ctx context.Context, client *AzureBlobClient, httpRequest *http.Request) error {
+	reqLogger := log.FromContext(ctx)
+	scopedLog := reqLogger.WithName("updateHttpRequestHeaderWithIAM")
 
-	// config := &Config{
-	// 	MaxRetries: 3,
-	// 	HTTPClient: httpClient,
-	// }
+	scopedLog.Info("Updating Azure Http Request with IAM")
 
-	// azureBlobClient, err = NewAzureSession(ctx, appAzureBlobEndPoint, storageAccountName, secretAccessKey, config)
-	// if err != nil {
-	// 	scopedLog.Info("Error creating new AzureBlob Client Session", "err", err)
-	// 	return nil
-	// }
+	// Create http request to retrive IAM oauth token from metadata URL
+	oauthRequest, err := http.NewRequest("GET", azureTokenFetchUrl, nil)
+	if err != nil {
+		scopedLog.Error(err, "Azure Blob Failed to create new token request")
+		return err
+	}
 
-	azureBlobClient = SplunkAzureBlobClientTemp{client: http.Client{}}
-	return &azureBlobClient
+	// Mark metadata flag
+	oauthRequest.Header.Set("Metadata", "true")
 
-}
+	// Create raw query for http request
+	values := oauthRequest.URL.Query()
+	values.Add("api-version", "2018-02-01")
+	values.Add("resource", "https://storage.azure.com/")
+	oauthRequest.URL.RawQuery = values.Encode()
 
-// GetAppsList get the list of apps from remote storage
-func (client *AzureBlobClient) GetAppsList(ctx context.Context) (RemoteDataListResponse, error) {
-	reqLogger := log.FromContext(ctx)
-	scopedLog := reqLogger.WithName("GetAppsList")
+	// Retrieve oauth token
+	resp, err := client.HttpClient.Do(oauthRequest)
+	if err != nil {
+		scopedLog.Error(err, "Azure blob,Errored when sending request to the server")
+		return err
+	}
+	defer resp.Body.Close()
 
-	scopedLog.Info("Getting Apps list", " S3 Bucket", client.BucketName, "Prefix", client.Prefix)
+	// Read http response
+	responseBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		scopedLog.Error(err, "Azure blob,Errored when reading resp body")
+		return err
+	}
 
-	azureBlobClient := client.Client
+	// Extract the token from the http response
+	var azureOauthTokenResponse TokenResponse
+	err = json.Unmarshal(responseBody, &azureOauthTokenResponse)
+	if err != nil {
+		scopedLog.Error(err, "Unable to unmarshal response to token")
+		return err
+	}
 
-	// Create a bucket list command for all files in bucket
+	// Update http request header with IAM access token
+	httpRequest.Header.Set("x-ms-version", "2017-11-09")
```

Comment:
Seems like an old version, would we want to update this to something newer as per https://docs.microsoft.com/en-us/rest/api/storageservices/previous-azure-storage-service-versions

Reply:
As per my understanding, any date should work here.